src/edu/csus/ecs/pc2/ui/SelectJudgementPaneNew.java

```diff
@@ -877,10 +877,14 @@ protected void executeRun() {
             Exception ex = executionData.getExecutionException();
             Language lang = getContest().getLanguage(run.getLanguageId());
 
+            //TODO: the following assignment and 'if' don't make sense (the assignment inside the 'if' is 
+            //     the same as the one preceding the 'if').   jlc
             String command = lang.getJudgeProgramExecuteCommandLine();
             if (command == null) {
                 command = lang.getJudgeProgramExecuteCommandLine();
             }
+            
+            //TODO:  the command line needs to be adjusted to reflect the (possible) presence of a sandbox.  jlc
```

Comment:
Fix this. This substitution should not be done here because the auto judge will never execute this code.
All substitutions should be done in Executable for that reason.

Reply:
This is not something I'm prepared to fix.  I'm not saying I disagree about where substitutions should be done.  I'm saying I didn't add the call to "substituteAllStrings()" in this code (which I believe is what you are objecting to).   The ONLY thing I added was the comment (TODO) pointing out that currently, SelectJudgementPaneNew is not coded to properly handle the presence of a sandbox.  I agree that it needs to be fixed.  I'm not at all sure what that fix should be; it's a design problem that I simply discovered.  Fixing it should be handled by creating a new Issue.

Reply:
I agree.   A new issue will be created to ensure that both AJ and UI use the same code/data to
execute then run.   I am pretty sure there is another part of the code with the same problem,
the AJ uses different code than the UI to read the validator output and create the judgement.
Issue #373 has been opened.